pkg/metrics/server.go

```diff
@@ -37,44 +44,137 @@ func RunServer(port int, stopCh <-chan struct{}) {
 		Handler: router,
 	}
 
-	go func() {
-		if err := srv.ListenAndServeTLS(tlsCRT, tlsKey); err != nil {
-			klog.Errorf("error starting metrics server: %v", err)
-		}
-	}()
-	<-stopCh
-	if err := srv.Shutdown(context.Background()); err != nil && err != http.ErrServerClosed {
-		klog.Errorf("error stopping metrics listener: %v", err)
+	return srv
+}
+
+func startServer(srv *http.Server, stopCh <-chan struct{}) {
+	klog.Infof("starting metrics server")
+	if err := srv.ListenAndServeTLS(tlsCert, tlsKey); err != nil && err != http.ErrServerClosed {
+		klog.Errorf("error from metrics server: %v", err)
 	}
 }
 
-// PodLabelsUsed indicates whether the deprecated Pod label matching functionality
-// is turned on.
-func PodLabelsUsed(enable bool) {
-	if enable {
-		podLabelsUsed.Set(1)
+func stopServer(srv *http.Server) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := srv.Shutdown(ctx); err != nil {
+		klog.Errorf("error or timeout stopping metrics listener: %v", err)
+	}
+	klog.V(2).Infof("stopping metrics server without error")
```

Comment:
stopping or stopped?  This could be my English though, so you're in a better position to asses which is better.

Reply:
You're definitely right this is awkward as written... It was fixed on my local copy :)